velox/expression/tests/ExprTest.cpp

```diff
@@ -1751,6 +1752,84 @@ TEST_F(ExprTest, ifWithConstant) {
   EXPECT_EQ(true, result->as<ConstantVector<bool>>()->valueAt(0));
 }
 
+// Make sure that switch do set nulls for rows that are not evaluated by the
+// switch due to a throw when the output type is not fixed primitive.
+TEST_F(ExprTest, switchSetNullsForThrowIndices) {
+  // Build an input with c0 column having nulls at odd row index.
+  registerFunction<TestingAlwaysThrowsFunction, bool, int64_t>(
+      {"always_throws"});
+  registerFunction<TestingThrowsAtOddFunction, bool, int64_t>({"throw_at_odd"});
+
+  auto eval = [&](const std::string& input, auto inputRow) {
+    auto exprSet = compileMultiple({input}, asRowType(inputRow->type()));
+
+    exec::EvalCtx context(execCtx_.get(), exprSet.get(), inputRow.get());
+    *context.mutableThrowOnError() = false;
+    std::vector<VectorPtr> results;
+    SelectivityVector rows(inputRow->size());
+    exprSet->eval(rows, context, results);
+    return results[0];
+  };
+
+  {
+    auto input = vectorMaker_.flatVector<int64_t>({1, 2, 3, 4, 5});
+    // Size of result will be less than rows.end() hence it will write nulls for
+    // errors.
+    auto result =
+        eval("if (always_throws(c0), 7, 1)", vectorMaker_.rowVector({input}));
+    for (int i = 0; i < input->size(); i++) {
+      EXPECT_TRUE(result->isNullAt(i));
+    }
+  }
+
+  {
+    // Size of result will be less than rows.end() hence it will write nulls for
+    // errors.
+    auto input = vectorMaker_.flatVector<int64_t>({1, 2, 3, 4, 5});
+    auto result =
+        eval("if(throw_at_odd(c0), 7, 1)", vectorMaker_.rowVector({input}));
+    EXPECT_EQ(result->size(), input->size());
+    for (int i = 0; i < input->size(); i += 2) {
+      EXPECT_TRUE(result->isNullAt(i)) << "index: " << i;
+    }
+  }
+
+  {
+    // Size of result = rows.end() hence it will NOT write nulls for
```

Comment:
do we need to update some of comments these that differentiate between primitive and complex types?
Also curious, do all these tests fail without this PR change?

Reply:
since i updated the code to set nulls also for primitives I also need to update the tests good catch